pkg/controller/core/clusterqueue_controller.go

```diff
@@ -129,18 +131,13 @@ func (r *ClusterQueueReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		if err := r.updateCqStatusIfChanged(ctx, newCQObj, metav1.ConditionTrue, "Ready", msg); err != nil {
 			return ctrl.Result{}, client.IgnoreNotFound(err)
 		}
-	} else if r.cache.ClusterQueueTerminating(newCQObj.Name) {
-		msg := "Can't admit new workloads; clusterQueue is terminating"
-		if err := r.updateCqStatusIfChanged(ctx, newCQObj, metav1.ConditionFalse, "Terminating", msg); err != nil {
-			return ctrl.Result{}, client.IgnoreNotFound(err)
-		}
 	} else {
-		msg := "Can't admit new workloads; some flavors are not found"
-		if err := r.updateCqStatusIfChanged(ctx, newCQObj, metav1.ConditionFalse, "FlavorNotFound", msg); err != nil {
+
+		reason, msg := r.cache.ClusterQueueInactiveReason(newCQObj.Name)
```

Comment:
I would actually go further and have a function `r.cache.ClusterQueueReadiness`, which returns (true/false, reason, message).

Reply:
Could you please clarify Why might we need a boolean here?

Reply:
see line 129 above. It is asking whether the CQ is active, which would get a lock, then release and then we would get another lock for CQInactiveReason.

All those operations can be merged into a single function that uses one lock.

Reply:
added function https://github.com/kubernetes-sigs/kueue/pull/1079/commits/dcdb80cb3c9ec7a61c936676fd8509bfd160f905